### PR TITLE
Make setup_auth recipe optional

### PR DESCRIPTION
**Pull request by asetiawan**
For agent/client installation, it might be useful to keep the default password, mainly for troubleshooting the forwarder connectivity. There isn't many security benefits gained from changing the agent's default password (especially since you can reset the password easily). For server installation, it makes sense to have the default password changed.

This pull request attempts to make setup_auth optional with non-breaking changes.

### CODE REVIEW

**Comment by Sauraus** [comment]
+1

**Comment by saimak** [comment]
When will we have this merged ? We too dont want setup_auth in forwarders. Thanks

File: attributes/default.rb [hunk 1]
```diff
@@ -21,6 +21,7 @@
 default['splunk']['web_port']       = '443'
 default['splunk']['ratelimit_kilobytessec'] = '2048'
 
+default['splunk']['change_splunk_auth'] = true
```
Comment (jtimberman) [inline attributes/default.rb]: Please rename the attribute to `node['splunk']['setup_auth']` to correspond to the recipe, making the intent more clear.

File: recipes/client.rb [hunk 2]
```diff
@@ -69,4 +69,7 @@
 end
 
 include_recipe 'chef-splunk::service'
-include_recipe 'chef-splunk::setup_auth'
+
+if node['splunk']['change_splunk_auth']
+  include_recipe 'chef-splunk::setup_auth'
```
Comment (jtimberman) [inline recipes/client.rb]: Use an inline conditional (using the renamed attribute per above comment):

```ruby
include_recipe 'chef-splunk::setup_auth' if node['splunk']['setup_auth']
```

**Comment by asetiawan** [comment]
Updated. Please review :smile:

**Comment by jtimberman** [comment]
I'm :+1:, but not actively working with this cookbook right now, so I won't have cycles to test and release anytime soon.

**Comment by jtimberman** [comment]
Found some cycles on a plane! :airplane: :rocket: :cake:

**Pull request merged**